lib/node_modules/@stdlib/array/complex64/lib/main.js

```diff
@@ -1755,6 +1755,85 @@ setReadOnly( Complex64Array.prototype, 'some', function some( predicate, thisArg
 	return false;
 });
 
+/**
+* Returns subarray elements.
+*
+* @name subarray
+* @memberof Complex64Array.prototype
+* @type {Function}
+* @param {integer} [begin] - starting index (inclusive)
+* @param {integer} [end] - ending index (non-inclusive)
+* @throws {TypeError} `this` must be a complex number array
+* @throws {TypeError} arguments must be integers
+* @returns {Complex64Array} subarray
+*
+* @example
+* var arr = new Complex64Array( 5 );
+*
+* arr.set( [ 1.0, -1.0 ], 0 );
+* arr.set( [ 2.0, -2.0 ], 1 );
+* arr.set( [ 3.0, -3.0 ], 2 );
+* arr.set( [ 4.0, -4.0 ], 3 );
+* arr.set( [ 5.0, -5.0 ], 4 );
+*
+* var subarr = arr.subarray();
+* // returns <Complex64Array>
+*
+* var len = subarr.length;
+* // returns 5
+*
+* subarr = arr.subarray( 1, -2 );
+* // returns <Complex64Array>
+*
+* len = subarr.length;
+* // returns 2
+*/
+setReadOnly( Complex64Array.prototype, 'subarray', function subarray( begin, end ) {
+	var buf;
+	var len;
+	var out;
+	var i;
+	if ( !isComplexArray( this ) ) {
+		throw new TypeError( 'invalid invocation. `this` is not a complex number array.' );
+	}
+	buf = this._buffer;
+	len = this._length;
+	if ( arguments.length === 0 ) {
+		begin = 0;
+		end = len;
+	} else {
+		if ( !isInteger( begin ) ) {
+			throw new TypeError( format( 'invalid argument. First argument must be an integer. Value: `%s`.', begin ) );
+		}
+		if ( begin < 0 ) {
+			begin += len;
+			if ( begin < 0 ) {
+				begin = 0;
+			}
+		}
+		if ( arguments.length === 1 ) {
+			end = len;
+		} else {
+			if ( !isInteger( end ) ) {
+				throw new TypeError( format( 'invalid argument. Second argument must be an integer. Value: `%s`.', end ) );
+			}
+			if ( end < 0 ) {
+				end += len;
+				if ( end < 0 ) {
+					end = 0;
+				}
+			} else if ( end > len ) {
+				end = len;
+			}
+		}
+	}
+	out = [];
+	for ( i = begin; i < end; i++ ) {
+		out.push( getComplex64( buf, i ) );
+	}
+	return new this.constructor( out );
```

Comment:
@Jaysukh-409 We can do better here. The creation of `Complex64` instances is unnecessary. Given that you know `begin` and `end`, you don't need to copy at all. In fact, here you are **not** returning a view. You are returning a new `Complex64Array` having an underlying array buffer which is different than the input array. That is not what is intended.

Instead, you can get rid of the loop and do sthg like

```
buf = this._buffer;
return new this.constructor( buf.buffer, buf.byteOffset*start*2*buf.BYTES_PER_ELEMENT, end-start-1 );
```

You should check that the above works, but the implementation should be sthg along these lines. Meaning, a view over the original buffer.

Reply:
@Jaysukh-409 One thing we need to guard against is when `end <= start`. For example, `arr.subarray( 2, 1 )`. In this case, we should return an empty view. In which case,

```
buf = this._buffer;
len = ( end-start ) * 2;
return new this.constructor( buf.buffer, buf.byteOffset + (start*2*buf.BYTES_PER_ELEMENT), ( len < 0 ) ? 0 : len );
```

We should add corresponding tests.